content/tutorial/tutorial.md

```diff
@@ -12,185 +12,189 @@ redirect_from:
 
-React is a declarative, efficient, and flexible JavaScript library for building user interfaces. It lets you compose complex UIs from small and isolated pieces of code called "components".
+उपयोगकर्ता इंटरफेस के निर्माण के लिए React एक वर्णनात्मक, कुशल और फ्लेक्सिबल जावास्क्रिप्ट लाइब्रेरी है। यह आपको "कौम्पोनॅन्ट्स" नामक कोड के छोटे और पृथक टुकड़ों से जटिल UI की रचना करने देती है।
 
-React has a few different kinds of components, but we'll start with `React.Component` subclasses:
+React में कुछ अलग-अलग प्रकार के कौम्पोनॅन्ट होते हैं, लेकिन हम `React.Component` उपवर्गों के साथ शुरुआत करेंगे:
 
 ```javascript
 class ShoppingList extends React.Component {
   render() {
     return (
       <div className="shopping-list">
-        <h1>Shopping List for {this.props.name}</h1>
+        <h1>{this.props.name} के लिए खरीदारी की सूची</h1>
         <ul>
-          <li>Instagram</li>
-          <li>WhatsApp</li>
-          <li>Oculus</li>
+          <li>इंस्टाग्राम</li>
+          <li>व्हाट्सैप्</li>
+          <li>ओकुलस</li>
         </ul>
       </div>
     );
   }
 }
 
-// Example usage: <ShoppingList name="Mark" />
+// उपयोग उदाहरण: <ShoppingList name="Mark" />
 ```
 
-We'll get to the funny XML-like tags soon. We use components to tell React what we want to see on the screen. When our data changes, React will efficiently update and re-render our components.
+हमें जल्दी ही अजीब XML जैसे टैग्स मिलेंगे. हम स्क्रीन पर जो देखना चाहते हैं उसे React करने के लिए हम कौम्पोनॅन्ट्स का उपयोग करते हैं। जब हमारा डेटा बदलता है, तो React कुशलतापूर्वक हमारे कौम्पोनॅन्ट्स को अद्यतन और प्रस्तुत करेगा।
 
-Here, ShoppingList is a **React component class**, or **React component type**. A component takes in parameters, called `props` (short for "properties"), and returns a hierarchy of views to display via the `render` method.
+यहाँ, ShoppingList एक **React कौम्पोनॅन्ट क्लास** या **React कौम्पोनॅन्ट टाइप** है। एक कौम्पोनॅन्ट मापदंडों में लेता है, जिसे `props` ("गुणों" के लिए छोटा) कहा जाता है, और `रंडर` मेथड के माध्यम से प्रदर्शित करने के लिए विचारों का एक पदानुक्रम देता है।
 
-The `render` method returns a *description* of what you want to see on the screen. React takes the description and displays the result. In particular, `render` returns a **React element**, which is a lightweight description of what to render. Most React developers use a special syntax called "JSX" which makes these structures easier to write. The `<div />` syntax is transformed at build time to `React.createElement('div')`. The example above is equivalent to:
+`render` मेथड स्क्रीन पर जो आप देखना चाहते हैं उसका *विवरण* देता है। React वर्णन लेती है और परिणाम प्रदर्शित करती है। विशेष रूप से, रेंडर एक **React तत्व** देता है, जो कि प्रस्तुत करना क्या है का एक हल्का विवरण है। अधिकांश React डेवलपर्स "JSX" नामक एक विशेष वाक्यविन्यास का उपयोग करते हैं जो इन संरचनाओं को लिखना आसान बनाता है। `<div />` सिंटैक्स को बिल्ड समय में `React.createElement ('div')` में बदल दिया जाता है। उपरोक्त उदाहरण इसके बराबर है:
```

Comment:
वर्णन would be better translation for description?

Reply:
देता -> deti

Reply:
तत्व -> don't translate

Reply:
जो कि प्रस्तुत करना क्या है का एक हल्का विवरण है। -> grammar check

Reply:
वाक्यविन्यास, संरचनाओं  -> too complex

Reply:
उपरोक्त -> upar diya gaya

Reply:
> जो कि प्रस्तुत करना क्या है का एक हल्का विवरण है। -> grammar check

जो की रेंडर का हल्का विवरण है